### PR TITLE
Minor: improve error message when ARRAY literals can not be planned

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

- Part of https://github.com/apache/datafusion/issues/13854
## Rationale for this change
There was a problem running substrait tests as they didn't activate the right features in DataFusion:

```
Error: Internal("Expected a simplified result, but none was found")
```

This error was confusing as it sounded like a bug and didn't hint at how to fix the problem

## What changes are included in this PR?
Improve the error message to be more specific about what is wrong and change from internal to not implemented

### CODE REVIEW

File: datafusion/sql/src/expr/value.rs [hunk 1]
```diff
@@ -169,7 +169,7 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
             }
         }
 
-        internal_err!("Expected a simplified result, but none was found")
+        not_impl_err!("Could not plan array literal. Hint: is `nested_expressions` DataFusion feature selected?")
```
Comment (comphead) [inline datafusion/sql/src/expr/value.rs:172]: I would say this question it is not clear a little bit. `nested_expressions` DataFusion feature selected - is it cause of issue or solution? 

```suggestion
        not_impl_err!("Could not plan array literal. Hint: Please try with `nested_expressions` DataFusion feature enabled")
```

Comment (alamb) [reply datafusion/sql/src/expr/value.rs:172]: That is a good call -- thank you @comphead

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

**Comment by alamb** [comment]
Thank you @comphead

**Pull request merged**